plugins/woo-ai/src/product-category/product-category-suggestions.tsx

```diff
@@ -0,0 +1,209 @@
+			}
+		},
+	} );
+
+	const handleSuggestionClick = ( suggestion: string ) => {
+		// remove the selected item from the list of suggestions
+		setSuggestions( suggestions.filter( ( s ) => s !== suggestion ) );
+		selectCategory( suggestion );
+
+		recordCategoryTracks( 'select', {
+			selected_category: suggestion,
+		} );
+	};
+
+	const buildPrompt = () => {
+		const allAvailableCategories = getAllAvailableCategories().join( ', ' );
+
+		const productPropsInstructions = generateProductDataInstructions();
+		const instructions = [
+			'You are a WooCommerce SEO and marketing expert.',
+			`Using the product's ${ productPropsInstructions.includedProps.join(
+				', '
+			) }, detect the main product category and its subcategories based on the existing available categories.`,
+			"Your goal is to enhance the store's SEO performance and sales.",
+			'Categories can have parents and multi-level children structures like Parent Category > Child Category.',
+			'You will be given a list of available categories and you can only return the best matching categories from this list.',
+			`Available categories are: ${ allAvailableCategories }`,
+			"The product's properties are:",
+			...productPropsInstructions.instructions,
+			'Return only a comma-separated list of the product categories, children categories must be separated by >.',
+			'Here is an example of a valid response:',
+			'Parent Category 1 > Subcategory 1, Parent Category 2 > Subcategory 2 > Another Subcategory 2',
+			'Do not output the example response. Respond only with the suggested categories. Do not say anything else.',
+		];
+
+		return instructions.join( '\n' );
+	};
+
+	const fetchProductSuggestions = async () => {
+		setSuggestions( [] );
+		setSuggestionsState( SuggestionsState.Fetching );
+
+		recordCategoryTracks( 'start', {
+			current_categories: getCategories(),
+		} );
+
+		await requestCompletion( buildPrompt() );
+	};
+
+	return (
+		<div className="wc-product-category-suggestions">
+			{ suggestions.length > 0 &&
+				suggestionsState !== SuggestionsState.Fetching && (
+					<SuggestionPills
+						suggestions={ suggestions }
+						onSuggestionClick={ handleSuggestionClick }
+					/>
+				) }
+			{ suggestionsState === SuggestionsState.Fetching && (
+				<div className="wc-product-category-suggestions__loading notice notice-info">
+					<p className="wc-product-category-suggestions__loading-message">
+						<RandomLoadingMessage
+							isLoading={
+								suggestionsState === SuggestionsState.Fetching
+							}
+						/>
+					</p>
+				</div>
+			) }
+			{ suggestionsState === SuggestionsState.NoMatch && (
+				<div className="wc-product-category-suggestions__no-match notice notice-success">
+					<p>
+						{ __(
+							'Your product is already well-categorized according to the properties you provided.',
+							'woocommerce'
+						) }
+					</p>
+				</div>
+			) }
+			{ suggestionsState === SuggestionsState.None && (
```

Comment:
I wonder if we could simplify this by combining "NoMatch and None" into a single "None" state. Ultimately, it seems like the same result: the user doesn't get a category suggestion, but for slightly different reasons. IMO we could combine them into a single state and make the returned message more generic, maybe like "Unable to generate a matching category for the product." 🤔

Reply:
Yeah that makes sense. I've removed the nomatch state in https://github.com/woocommerce/woocommerce/pull/39437/commits/d1c28b12f0662fe38e44ff464004d8c615ddcf41